app/helpers/i18n_helper.rb

```diff
@@ -47,6 +47,37 @@ def date_eras_gregorian(date)
     nil
   end
 
+  # Return the most accurate date from an array of dates using scoring algorithm below
+  def date_most_accurate(dates)
+    return nil unless dates.present?
+    return dates if dates.is_a?(String)
```

Comment:
this is still not formatting `dates` even if we were allowing `date_most_accurate` to be called with a string as a parameter, then we still need to format that string as gregorian.